syft/pkg/cataloger/common/cpe/field_candidate.go

```diff
@@ -56,6 +56,15 @@ func (s fieldCandidateSet) removeByValue(values ...string) {
 	}
 }
 
+// removeByCondition removes all entries from the fieldCandidateSet for which the condition function returns true.
+func (s fieldCandidateSet) removeByCondition(condition func(candidate fieldCandidate) bool) {
```

Comment:
you may have considered this already , so disregard if so; there is already a filter function with the same signature for `fieldCandidates`: https://github.com/anchore/syft/blob/main/syft/pkg/cataloger/common/cpe/field_candidate_filter.go#L4 . Could we use that signature here and keep similar verbiage (either renaming the existing `filter` to `condition` or the other way around)?

Reply:
Great suggestion! I hadn't considered that. I'll unify the approach here...

Reply:
Just pushed a couple of commits to address this. I used the existing function type, and I used "condition" as the noun. This seemed to work well, since it implies only what it is — that something about a `fieldCandidate` might be true or false. It's up to the caller to decide what to do with that information (such as remove an item from a set based on this condition).

A portion of these changes _may_ be superseded via upcoming work to address https://github.com/anchore/syft/pull/514#discussion_r715096621.